crates/phactory/src/contracts/pink.rs

```diff
@@ -503,12 +514,23 @@ impl Cluster {
                 transfer,
                 estimating,
             } => {
+                let origin = origin.cloned().ok_or(QueryError::BadOrigin)?;
                 let _guard = context
                     .query_scheduler
                     .acquire(contract_id.clone(), context.weight)
                     .await
                     .or(Err(QueryError::ServiceUnavailable))?;
 
+                if let Some(logger) = &context.log_handler {
+                    let salt = self.default_runtime().get_key();
+                    let fp = blake2_128(&(b"finger print:", &origin, salt).encode());
```

Comment:
Maybe we can make it even shorter. Probably twox_64 is enough. It's also faster though.

Reply:
Sounds OK.
Done.

Reply:
Oh, sorry. I think XXHash is not a cryptographically secure hash. So I think need to ensure `salt` is not exactly the cluster key, but a cryptographic hash should be used to produce the `salt`.

Reply:
Sounds good

Reply:
> Oh, sorry. I think XXHash is not a cryptographically secure hash. So I think need to ensure `salt` is not exactly the cluster key, but a cryptographic hash should be used to produce the `salt`.

Oh, I thought you said `blake2_64`, so I didn't use `twox_64` but `blake2_64`. Do you think it's OK?

Reply:
`blake2_64` is fine. It would be better if we can use `blake2` to generate the salt (potentially cached in memory) and use `xxhash` to produce fingerprint, since xxhash is much cheaper than blake.

Reply:
Btw, it would be nice to document our methodology somewhere.

Reply:
> It would be better if we can use blake2 to generate the salt (potentially cached in memory) and use xxhash to produce fingerprint, since xxhash is much cheaper than blake.

Done

> Btw, it would be nice to document our methodology somewhere.

Added `docs/worker-statistics-api.md`